toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1421,17 +1437,24 @@ func InstallBootloader(installChroot *safechroot.Chroot, encryptEnabled bool, bo
 // as part of a general "root" partition is assumed to have been done already.
 func installLegacyBootloader(installChroot *safechroot.Chroot, bootDevPath string) (err error) {
 	const (
-		squashErrors = false
+		squashErrors     = false
+		bootDir          = "/boot"
+		bootDirArg       = "--boot-directory"
+		targetArg        = "--target"
+		legacyTargetArch = "i386-pc"
+		grub2BootDir     = "/boot/grub2"
 	)
 
-	// Since we do not have grub2-pc installed in the setup environment, we need to generate the legacy grub bootloader
-	// inside of the install environment. This assumes the install environment has the grub2-pc package installed
-	err = installChroot.UnsafeRun(func() (err error) {
-		err = shell.ExecuteLive(squashErrors, "grub2-install", "--target=i386-pc", "--boot-directory=/boot", bootDevPath)
-		err = shell.ExecuteLive(squashErrors, "chmod", "-R", "go-rwx", "/boot/grub2/")
-		return
-	})
+	grub2InstallTargetArg := fmt.Sprintf("%s=%s", targetArg, legacyTargetArch)
```

Comment:
Suggestion: I'm a strong supporter for using well-named consts to avoid repetitions, "magic numbers" or not-immediately-clear arguments (the `squashErrors` vs a plain `false` case for instance). In this case, however, I think that we can directly use `--target=i386-pc` inside `shell.ExecuteLive`, since we're using them only once and the strings themselves provide the same amount of information as the constants' names, so we're not losing clarity. We might actually increase it, since the reader won't have to take the additional 2 steps of checking what `grub2InstallTargetArg` is and then look at the consts.

Similarly, I think we can just go ahead and skip the `bootDirArg` const and use its value directly.

Reply:
Going to keep `bootDirArg` as an Sprintf since the `installBootDir` is `installChroot.RootDir()/boot`, not just a const.